connectorx-python/connectorx/__init__.py

```diff
@@ -247,6 +247,66 @@ def read_sql(
 ) -> pl.DataFrame: ...
 
 
+def get_passfile_content(path: pathlib.Path) -> dict:
+    # host:port:db_name:user_name:password
+    with open(path) as f:
```

Comment:
How about we do something different? the content we need from pgpass is the password only the restof the elements are used for retrieving it. So:

```python
            for line in f:
                entry = read_pgpass_line(line)
                if entry:
                    if ((entry['host'] == host or entry['host'] == '*') and
                        (entry['port'] == port or entry['port'] == '*') and
                        (entry['database'] == database or entry['database'] == '*') and
                        (entry['user'] == user or entry['user'] == '*')):
                        return entry['password']
```